src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,185 @@
+     * The dates include each day to be monitored in a 7-day window starting from the trip's query start date.
+     * @return a list of date strings in YYYY-MM-DD format corresponding to each day of the week to monitor.
+     */
+    public static List<String> getDatesToCheckItineraryExistence(MonitoredTrip trip) throws URISyntaxException {
+        List<String> result = new ArrayList<>();
+        ZoneId zoneId = trip.tripZoneId();
+        Map<String, String> params = getQueryParams(trip.queryParams);
+
+        // Start from the query date, if available.
+        // If there is no query date, start from today.
+        String queryDateString = params.getOrDefault(DATE_PARAM, DateTimeUtils.getStringFromDate(DateTimeUtils.nowAsLocalDate(), YYYY_MM_DD));
+        LocalDate queryDate = DateTimeUtils.getDateFromString(queryDateString, YYYY_MM_DD);
+
+        // Get the trip time.
+        // FIXME: replace this block with code from PR #75.
+        String[] tripTimeSplit = params.get("time").split(":");
+        int tripHour = Integer.parseInt(tripTimeSplit[0]);
+        int tripMinutes = Integer.parseInt(tripTimeSplit[1]);
+        ZonedDateTime queryZonedDateTime = ZonedDateTime.of(queryDate, LocalTime.of(tripHour, tripMinutes), zoneId);
+
+
+        // Check the dates in a 7-day window starting from the query date.
+        for (int i = 0; i < 7; i++) {
+            ZonedDateTime probedDate = queryZonedDateTime.plusDays(i);
+            if (trip.isActiveOnDate(probedDate)) {
+                result.add(DateTimeUtils.getStringFromDate(probedDate.toLocalDate(), YYYY_MM_DD));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets OTP queries to check non-realtime itinerary existence for the given trip.
+     */
+    public static Map<String, String> getItineraryExistenceQueries(MonitoredTrip trip) throws URISyntaxException {
+        return getQueriesFromDates(
+            excludeRealtime(trip.queryParams),
+            getDatesToCheckItineraryExistence(trip)
+        );
+    }
+
+    /**
+     * @return a copy of the specified query, with ignoreRealtimeUpdates set to true.
+     */
+    public static String excludeRealtime(String queryParams) throws URISyntaxException {
+        Map<String, String> params = getQueryParams(queryParams);
+        params.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+
+        // Insert '?' so others can parse the resulting query string.
+        return "?" + toQueryString(params);
+    }
+
+    /**
+     * Replaces the trip itinerary field value with one from the verified itineraries queried from OTP.
+     * The ui_activeItinerary query parameter is used to determine which itinerary to use,
+     * TODO/FIXME: need a trip resemblance check to supplement the ui_activeItinerary param used in this function.
+     */
+    public static void updateTripWithVerifiedItinerary(MonitoredTrip trip, List<Itinerary> verifiedItineraries) throws URISyntaxException {
```

Comment:
This method can result in a completely different itinerary being saved than what the user selected. We should discuss things a little more so that we're on the same page of what to do when a user wants to save an itinerary that is only made possible due to unique realtime data. For example, say a bus A is running early and is able to get to a bus stop just before Bus B leaves because Bus B is late. That result won't ever show up in these additional checks here. We probably need some kind of way to determine if an itinerary is typically possible at all or was made possible by the current day's realtime conditions.